build-aux/snap/snapcraft.yaml

```diff
@@ -51,6 +49,17 @@ parts:
       sudo -E apt-get build-dep -y ./
       ./get-deps.sh --skip-unused-check
     override-build: |
+      # unfortunately we do not have something like "snapcraftctl get-version"
+      # so we have to re-run mkversion.sh to check if the version number was set
+      # as "dirty" from the override-pull step
+      if sh -x ./mkversion.sh --output-only | grep "dirty"; then
```

Comment:
could we just save the version to some file in the previous step?

Reply:
I don't think it takes a lot of time or resources to run the script, IMHO it's just slightly inelegant that we can't just ask snapcraft what we set the version to already with `snapcraftctl get-version`, since we already told it the version number with `snapcraftctl set-version`.

Reply:
I mean there's probably an assumption that if you called set-version you know what version it is.